docs/csharp/programming-guide/classes-and-structs/snippets/destructors/Program.cs

```diff
@@ -0,0 +1,48 @@
+﻿//<Snippet1>
+class First
+{
+    ~First()
+    {
+        System.Diagnostics.Trace.WriteLine("First's destructor is called.");
+    }
+}
+
+class Second : First
+{
+    ~Second()
+    {
+        System.Diagnostics.Trace.WriteLine("Second's destructor is called.");
+    }
+}
+
+class Third : Second
+{
+    ~Third()
+    {
+        System.Diagnostics.Trace.WriteLine("Third's destructor is called.");
+    }
+}
+
+class TestDestructors
+{
+    static void Main()
+    {
+        Third t = new Third();
+    }
+}
+/* Output (to VS Output Window):
+    Third's destructor is called.
+    Second's destructor is called.
+    First's destructor is called.
+*/
```

Comment:
I couldn't get this output from the command-line using `dotnet run`.

Reply:
That's not surprising. There have been some changes in the exit guarantees for programs and executing finalizers starting with .NET Core.

Reply:
@BillWagner Do you want to file an issue for this article to update it and making this more clear?

Reply:
Yes. Thank you @Youssef1313 The changes are covered in the standard section on the GC, if I recall correctly.

Reply:
@BillWagner Filed #22513